config/initializers/_service_providers.rb

```diff
@@ -2,3 +2,7 @@
                     fetch(Rails.env, {})
 
 VALID_SERVICE_PROVIDERS = JSON.parse(Figaro.env.valid_service_providers)
+
+# merge from dashboard
+require 'feature_management'
+ServiceProviderUpdater.new.run if FeatureManagement.use_dashboard_service_providers?
```

Comment:
You are reading the code correctly, @zachmargolis 

The dilemma at present is that `SERVICE_PROVIDERS` is an in-memory global hash, so a background async job would have to do some gymnastics in order to update the running web app.

The best solution, imo, is to migrate to an active ActiveRecord-backed `ServiceProvider` model to persist storage beyond app restart, which would also allow us to flag (or eliminate) the `VALID_SERVICE_PROVIDERS` variable as well.

This solution is only ever intended to be run in non-production environments, and gated with the feature flag.

After discussing with @amoose we agreed on the AR approach but because it would involve some significant surgery, opted to do the MVP instead for now.